### PR TITLE
[MTurk] Better hit status tracking via complete post

### PR DESCRIPTION
**Pull request by JackUrb**
**Patch description**
Back in the day, we used to rely solely on hit status checks that pinged amazon's server directly to tell whether or not an assignment was completed. This behavior was wasteful, so we moved to SNS messages to tell us when a HIT had been submitted. These still had delays and a chance of being missed.

In #1610 I introduced static tasks, which relied on HIT triggering a POST event to determine when a HIT was submitted. This PR applies the same flow to regular tasks, and removes the old flow for checking the completion status of a HIT.

**Changes summary**
`Agents`: Created a `completed_act` which stores the value POSTed when a hit is completed on the MTurk side. Setters and getters followed. Removed all mentions of `get_agent_work_status` and replaced with checks against `hit_is_complete`.

`data_model`, `server.js`, `socket_manager`: Renamed `STATIC_MESSAGE` to `SUBMIT_MESSAGE` to reflect the new behavior of this flag. Similar renames around this idea.

`react_server/dev`: Merged `staticAllDoneCallback` and `allDoneCallback` as they now have the same behavior.

`WorkerManager`: special cases around the `SUBMIT_MESSAGE` to mark an agent as done when received.

**Testing steps**
Run `qa_data_collection`, ensure hits get marked complete and the manager cleans up.
Run a static task, ensure the same.

### CODE REVIEW

File: parlai/mturk/core/dev/agents.py [hunk 1]
```diff
@@ -161,15 +159,16 @@ def __init__(self, opt, mturk_manager, hit_id, assignment_id, worker_id):
         self.worker_id = worker_id
         self.some_agent_disconnected = False
         self.hit_is_expired = False
-        self.hit_is_abandoned = False  # state from Amazon MTurk system
-        self.hit_is_returned = False  # state from Amazon MTurk system
-        self.hit_is_complete = False  # state from Amazon MTurk system
+        self.hit_is_abandoned = False  # state from MTurk SNS system
```
Comment (stephenroller) [inline parlai/mturk/core/dev/agents.py]: what is SNS

File: parlai/mturk/core/dev/agents.py [hunk 2]
```diff
@@ -298,16 +297,20 @@ def reduce_state(self):
         self.msg_queue = None
         self.recieved_packets = None
 
-    def get_new_act_message(self):
-        """Get a new act message if one exists, return None otherwise"""
+    def check_disconnects(self):
+        """Ensures that an agent is still connected"""
```
Comment (stephenroller) [inline parlai/mturk/core/dev/agents.py]: Kinda weird to have the docstring inverted from the name of the method (check disconnects, check if connected)

Comment (stephenroller) [reply parlai/mturk/core/dev/agents.py]: Maybe assert_connected?

File: parlai/mturk/core/dev/agents.py [hunk 3]
```diff
@@ -515,24 +517,13 @@ def wait_completion_timeout(self, iterations):
...
-            start_time = time.time()
         wait_periods = 1
         self.wait_completion_timeout(wait_periods)
         sync_attempts = 0
-        while (
-            not self.hit_is_complete
-            and self.m_get_agent_work_status(self.assignment_id) != self.ASSIGNMENT_DONE
-        ):
+        while not self.hit_is_complete:
             if sync_attempts < 8:
```
Comment (stephenroller) [inline parlai/mturk/core/dev/agents.py]: magic constant

File: parlai/mturk/core/dev/react_server/server/server.js [hunk 4]
```diff
@@ -274,7 +274,7 @@ class LocalAgentState {
 // ======================= <Socket> =======================
 
 // Socket function types
-// TODO move to constants
+// WISH move to constants
```
Comment (stephenroller) [inline parlai/mturk/core/dev/react_server/server/server.js]: lol you being honest with yourself?

Comment (stephenroller) [reply parlai/mturk/core/dev/react_server/server/server.js]: i'm not sure WISH is in the common list of editor highlights/greps

https://github.com/JohnPostlethwait/fixme

File: parlai/mturk/core/dev/worker_manager.py [hunk 5]
```diff
@@ -141,8 +142,13 @@ def route_packet(self, pkt):
                 'Manager received: {}'.format(pkt),
                 should_print=self.opt['verbose'],
             )
-            # Push the message to the agent
-            agent.put_data(pkt.id, pkt.data)
+            # WISH worker_manager shouldn't need to know packet types
```
Comment (stephenroller) [inline parlai/mturk/core/dev/worker_manager.py]: ditto

File: parlai/mturk/core/dev/react_server/server/server.js (hunk 4 above)
Comment (JackUrb) [reply parlai/mturk/core/dev/react_server/server/server.js]: TIL the common codetags

File: parlai/mturk/core/dev/agents.py (hunk 1 above)
Comment (JackUrb) [reply parlai/mturk/core/dev/agents.py]: Simple Notification Service. It's now the only place we get pinged about someone who lets a HIT expire without completing it.

File: parlai/mturk/core/dev/agents.py (hunk 3 above)
Comment (JackUrb) [reply parlai/mturk/core/dev/agents.py]: Turns out we don't need most of this block anymore

**Pull request merged**